src/swc_util.rs

```diff
@@ -161,6 +161,47 @@ impl AstParser {
     }
   }
 
+  pub(crate) fn parse_script(
+    &self,
+    file_name: &str,
+    syntax: Syntax,
+    source_code: &str,
+  ) -> (
+    Result<swc_ecmascript::ast::Script, SwcDiagnosticBuffer>,
+    SingleThreadedComments,
+  ) {
+    let swc_source_file = self.source_map.new_source_file(
+      FileName::Custom(file_name.to_string()),
+      source_code.to_string(),
+    );
+
+    let buffered_err = self.buffered_error.clone();
+
+    let comments = SingleThreadedComments::default();
+    let lexer = Lexer::new(
+      syntax,
+      JscTarget::Es2019,
+      StringInput::from(&*swc_source_file),
+      Some(&comments),
+    );
+
+    let mut parser = Parser::new_from(lexer);
```

Comment:
This snippet is the same as `parse_module`, right? If so, it would be more readable if we extracted it into a common function or something IMO.

Reply:
Yeah, not sure yet, let's keep it that way for now